### PR TITLE
Reintroduce Cross Context Dispatch in Jetty 12

### PR DESCRIPTION
**Pull request by janbartel**
Implement cross context dispatch.

### CODE REVIEW

**Comment by joakime** [comment]
This PR is on track for 12.0.8 release, right?

**Comment by gregw** [comment]
> This PR is on track for 12.0.8 release, right?

Hopefully yes.   It might not be complete there, but we should have most of the common use-cases covered.

**Comment by gregw** [comment]
@joakime if you would like to help on this PR, at some stage we need an integration test that checks all the possible combinations of:
 + (ee8|ee9|ee10) ---FWD--> (ee8|ee9|ee10), 
 + (ee8|ee9|ee10) ---INC--> (ee8|ee9|ee10), 
 + intra FWD before / after cross context
 + intra INC before / after cross context

etc

**Comment by gregw** [comment]
@joakime @lachlan-roberts @olamy This is a big PR that we want in this release cycled (which may be rushed).   Can you start reviewing this ASAP so it makes it in.

The key areas to consider are:
 + How we find other contexts
 + How we wrap other contexts as ServletContexts
 + How we create a special RequestDispatcher from the wrapped cross context ServletContext
 + The wrapping done before dispatch to turn eeX request/response back into core request/response
 + The slight changes to wrapping done in ContextHandler and/or ServletContentHandler to handle cross context dispatch.
 + Cross context params work because of the request attribute used by Fields.from, even if EE8/9 do their own parsing, they now use the Fields data structure
 + Multipart params are a bit limited, specifically as the data structure is different, we can't do cross environment dispatch
 + Cross context sessions work because of cross context cookies and and shared session id manager
 + Cross context bypasses security handler

@joakime any thoughts on better testing all of this?

**Comment by joakime** [comment]
PR #11516 is needed for the testing additions I'm working on.

**Comment by olamy** [comment]
this is passing TCK. Need some Arquillian changes though. https://github.com/arquillian/arquillian-container-jetty/pull/186

**Review by joakime — DISMISSED** [review]
Merge this, it is quite good as it currently stands.
We'll keep adding testing it and refining over time.

File: jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/RequestTest.java [hunk 1]
```diff
@@ -339,6 +339,7 @@ protected void service(HttpServletRequest request, HttpServletResponse resp)
             assertThat(connection.getResponse(), containsString(" 200 OK"));
         }
 
+        cookieHistory.forEach(System.err::println);
```
Comment (janbartel) [inline jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/RequestTest.java]: Should remove this System.err.println

**Review by joakime — APPROVED** [review]
Lets get this merged.  It's a good change.

**Pull request merged**